infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/utils/CorfuSaasEndpointProvider.java

```diff
@@ -0,0 +1,64 @@
+package org.corfudb.infrastructure.logreplication.infrastructure.utils;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * Helper methods to retrieve the corfu's SaaS endpoint.
+ */
+@Slf4j
+public final class CorfuSaasEndpointProvider {
+    private static CorfuSaasEndpointProvider instance;
+    
+    private final Optional<String> corfuSaasEndpoint;
+
+    private final boolean isSaasDeployment;
+
+    private CorfuSaasEndpointProvider(String configFile, boolean isSaasDeployment) {
+        this.isSaasDeployment = isSaasDeployment;
+        this.corfuSaasEndpoint = getCorfuSaasEndpoint(configFile);
+    }
+
+    public static void init(String pluginConfigFile, boolean isSaasDeployment) {
+        instance = new CorfuSaasEndpointProvider(pluginConfigFile, isSaasDeployment);
+    }
+
+    private Optional<String> getCorfuSaasEndpoint(String pluginConfigFilePath) {
+        // if not SaaS deployment, return an empty optional
+        if(!isSaasDeployment) {
+            log.debug("Not a SaaS deployment");
+            return Optional.empty();
+        }
+
+        if (pluginConfigFilePath != null) {
+            return extractSaasEndpoint(pluginConfigFilePath);
+        }
+        log.warn("plugin config file not found {}", pluginConfigFilePath);
+        return Optional.empty();
+    }
+
+    private Optional<String> extractSaasEndpoint(String path) {
+        try (InputStream input = new FileInputStream(path)) {
+            Properties prop = new Properties();
+            prop.load(input);
+            String endpoint = prop.getProperty("saas_endpoint");
+            return Optional.ofNullable(endpoint);
+        } catch (IOException e) {
+            log.warn("Error extracting corfu saas endpoint");
+            return Optional.empty();
+        }
+    }
+
+    /**
+     * Get SaaS endpoint for corfu
+     * @return endpoint when deployment is saas, otherwise return null
+     */
+    public static Optional<String> getCorfuSaasEndpoint() {
```

Comment:
Since this method and the one invoked from the constructor have the same signature, they are being considered overloaded.  Can we can rename one of the get() methods?  Or combine the get() and extract() methods above?

Reply:
done.